test/gtest/plugin_manager.cpp

```diff
@@ -70,6 +70,60 @@ class LoadMultiplePluginsTestFixture
   }
 };
 
+class LoadedPluginTestFixture : public testing::Test {
+protected:
+  /* Plugin Manager. */
+  nixlPluginManager &plugin_manager_ = nixlPluginManager::getInstance();
+  /* Static plugins. */
+  std::set<std::string> static_plugins_;
+  /* Dynamically loaded plugins. */
+  std::set<std::string> loaded_plugins_;
+
+  void SetUp() override {
+    for (const auto &plugin : nixlPluginManager::getStaticPlugins()) {
+      static_plugins_.insert(plugin.name);
+    }
+  }
+
+  void TearDown() override {
+    for (const auto &plugin : loaded_plugins_) {
+      plugin_manager_.unloadPlugin(plugin);
+    }
+  }
+
+  /*
+   * Load a plugin to plugin manager.
+   *
+   * Returns true if the plugin loaded succesfully, otherwise false.
+   */
+  bool LoadPlugin(std::string name) {
+    loaded_plugins_.insert(name);
+    return plugin_manager_.loadPlugin(name) != nullptr;
+  }
```

Comment:
Wonder if these two functions should be just ctor/dtor as per https://google.github.io/googletest/faq.html#CtorVsSetUp